forge/src/gas_report.rs

```diff
@@ -58,7 +61,11 @@ impl GasReport {
             let report_for = self
                 .report_for
                 .iter()
-                .any(|s| s == name.rsplit(':').next().unwrap_or(name.as_str()));
+                .any(|s| s == name.rsplit(':').next().unwrap_or(name.as_str())) &
```

Comment:
We can replace the double iterators with something like:

```rs
let contract_name = name.rsplit(':').next().unwrap_or(name.as_str());
let report_for = self.report_for.contains(contract_name) && !self.ignore.contains(contract_name);
```

Reply:
Whoops, you're right. Just committed what you suggested, with a little change to satisfy the argument type of `.contains()`.